pycsob/client.py

```diff
@@ -17,6 +19,150 @@ def send(self, request, **kwargs):
 
 
+@dataclass
+class CartItem:
+    """Cart item for creating card payment."""
+    name: str
+    quantity: int
+    amount: int
+    description: Optional[str] = None
+
+    def to_dict(self) -> OrderedDict[str, Any]:
+        """Transform to ordered dictionary."""
+        pairs = (
+            ("name", self.name[:20]),
+            ("quantity", self.quantity),
+            ("amount", self.amount),
+            ("description", self.description[:40] if self.description is not None else None),
+        )
+        return OrderedDict([(k, v) for k, v in pairs if v not in conf.EMPTY_VALUES])
+
+
+@dataclass
+class CustomerData:
+    """Customer data for creating card payment."""
+    name: str
+    email: Optional[str] = None
+    home_phone: Optional[str] = None
+    work_phone: Optional[str] = None
+    mobile_phone: Optional[str] = None
+
+    # Account data
+    account_created_at: Optional[str] = None
+    account_changed_at: Optional[str] = None
+    account_changed_pwd_at: Optional[str] = None
+    account_order_history: Optional[int] = None
+    account_payments_day: Optional[int] = None
+    account_payments_year: Optional[int] = None
+    account_one_click_adds: Optional[int] = None
+    account_suspicious: Optional[bool] = None
+
+    # Login data
+    login_auth: Optional[str] = None
+    login_auth_at: Optional[str] = None
+    login_auth_data: Optional[str] = None
+
+    def to_dict(self) -> OrderedDict[str, Any]:
+        """Transform to ordered dictionary."""
+        account_pairs = (
+            ("createdAt", self.account_created_at),
+            ("changedAt", self.account_changed_at),
+            ("changedPwdAt", self.account_changed_pwd_at),
+            ("orderHistory", self.account_order_history),
+            ("paymentsDay", self.account_payments_day),
+            ("paymentsYear", self.account_payments_year),
+            ("oneclickAdds", self.account_one_click_adds),
+            ("suspicious", self.account_suspicious),
+        )
+        account_dict = OrderedDict([(k, v) for k, v in account_pairs if v not in conf.EMPTY_VALUES])
+        login_pairs = (
+            ("auth", self.login_auth),
+            ("authAt", self.login_auth_at),
+            ("authData", self.login_auth_data),
+        )
+        login_dict = OrderedDict([(k, v) for k, v in login_pairs if v not in conf.EMPTY_VALUES])
+        pairs = (
+            ("name", self.name[:45]),
+            ("email", self.email[:100] if self.email else None),
+            ("homePhone", self.home_phone),
+            ("workPhone", self.work_phone),
+            ("mobilePhone", self.mobile_phone),
+            ("account", account_dict),
+            ("login", login_dict),
+        )
+        return OrderedDict([(k, v) for k, v in pairs if v not in conf.EMPTY_VALUES])
+
+
+@dataclass
+class OrderAddress:
+    """Order address (billing or shipping)."""
+    address_1: str
+    city: str
+    zip: str
+    country: str
+    address_2: Optional[str] = None
```

Comment:
The order of attributes should follow the documentation.

Reply:
In my opinion, it is better to follow the order according to the documentation, even though there are optional items before the mandatory ones. I don't think it matters in any way. Named attributes are used when entering.